ksql-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java

```diff
@@ -147,10 +148,15 @@ public String visitString(final Schema schema) {
 
     @Override
     public String visitBytes(final Schema schema) {
-      DecimalUtil.requireDecimal(schema);
-      return "DECIMAL("
-          + DecimalUtil.precision(schema) + ", "
-          + DecimalUtil.scale(schema) + ")";
+      if (DecimalUtil.isDecimal(schema)) {
+        return "DECIMAL("
+            + DecimalUtil.precision(schema) + ", "
+            + DecimalUtil.scale(schema) + ")";
+      } else if (GenericsUtil.isGeneric(schema)) {
+        return GenericsUtil.name(schema);
```

Comment:
This still feels wrong :(

Maybe the UDF code should have its own impl of `SqlSchemaFormatter`.  It could just extend this impl and add the generics handling bit.

Reply:
Did you suggestion